tests/unit_test/test_ga_etl_state.py

```diff
@@ -0,0 +1,72 @@
+from unittest.mock import patch
+import pytest
+from botocore.exceptions import ClientError
+from data_pipeline.google_analytics import etl_state as etl_state_module
+from data_pipeline.google_analytics.etl_state import (
+    get_stored_state,
+)
+from data_pipeline.google_analytics.ga_config import GoogleAnalyticsConfig
+
+
+GA_CONFIG = {
+    'gcpProjectName': 'elife-data-pipeline',
+    'dataPipelineId': 'ga_pipeline_id',
+    'dataset': 'dataset'
+}
+
+
+@pytest.fixture(name="mock_download_s3_json_object_exception")
+def _download_s3_json_object_with_exception():
```

Comment:
I would make this fixture get "mock_download_s3_json_object" as an input and set the side_effect. But it may be better to just set the side_effect as part of the test itself. You could still have a method that returns the ClientError. e.g.

```python
mock_download_s3_json_object.side_effect = get_no_such_key_client_error()
```

Alternatively that could even be a constant, e.g. NO_SUCH_KEY_CLIENT_ERROR